core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -347,6 +348,8 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
     def writeNextInputToStream(dataOut: DataOutputStream): Boolean
 
     def open(dataOut: DataOutputStream): Unit = Utils.logUncaughtExceptions {
+      val isUnixDomainSock = authHelper.conf.get(PYTHON_UNIX_DOMAIN_SOCKET_ENABLED)
```

Comment:
Instead of if/else's everywhere you could consider using an interface with two implementations. That improves readability of the code by a lot...

Reply:
Yup I will do it in a separate PR that factors those out. I think we can take the common code out to utils.